src/App.js

```diff
@@ -14,11 +14,90 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends Component {
+  state = {
+    goods: goodsFromServer,
+    showList: false,
+  }
+
+toggleListHandler = () => {
+  this.setState(state => ({
+    showList: !state.showList,
+  }));
+};
+
+reverseList = () => {
+  this.setState(state => ({
+    goods: [...state.goods].reverse(),
+  }));
+};
+
+sortList = () => {
+  this.setState(state => ({
+    goods: [...state.goods].sort(),
+  }));
+}
+
+resetList = () => {
+  this.setState({
+    goods: goodsFromServer,
+  });
+};
+
+sortByLength = () => {
+  this.setState(state => ({
+    goods: [...state.goods].sort((a, b) => a.length - b.length),
+  }));
+};
+
+render() {
+  return (
+    <div className="app">
+      <h1>Goods</h1>
+      <button
+        className={this.state.showList && 'visibility'}
+        type="button"
+        onClick={this.toggleListHandler}
+      >
+        Start
+      </button>
+      <button
+        className={!this.state.showList && 'visibility'}
+        type="button"
+        onClick={this.reverseList}
+      >
+        Reverse
+      </button>
+      <button
+        className={!this.state.showList && 'visibility'}
+        type="button"
+        onClick={this.sortList}
+      >
+        Sort
+      </button>
+      <button
+        className={!this.state.showList && 'visibility'}
+        type="button"
+        onClick={this.resetList}
+      >
+        Reset
+      </button>
+      <button
+        className={!this.state.showList && 'visibility'}
+        type="button"
+        onClick={this.sortByLength}
+      >
+        Sort by length
+      </button>
+      <ul>
+        {this.state.showList && this.state.goods.map(product => (
+          <li key={product}>{product}</li>
```

Comment:
Let's use a component approach and create a separate component `GoodsList`.

Also, it would be great to move all the sortings logic inside the `GoodsList` component.
It's good to split the logic. `App` component should only have the button 'start' which will toggle the render of the `GoodsList`. There you should pass initial goods and sort them inside that component (not inside the `App`)